pkg/server/metrics.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2025 The Sigstore Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"context"
+	"errors"
+	"log/slog"
+	"net"
+	"net/http"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+	"time"
+
+	grpc_prometheus "github.com/grpc-ecosystem/go-grpc-middleware/providers/prometheus"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"sigs.k8s.io/release-utils/version"
+)
+
+type metrics struct {
+	reg           *prometheus.Registry
+	serverMetrics *grpc_prometheus.ServerMetrics
+	// metrics
+	newHashedRekordEntries prometheus.Counter
+	newDsseEntries         prometheus.Counter
+	httpLatency            *prometheus.HistogramVec
+	httpRequestsCount      *prometheus.CounterVec
+}
+
+// Metrics provides the singleton metrics instance
+func getMetrics() *metrics {
+	return _initMetricsFunc()
+}
+
+var _initMetricsFunc = sync.OnceValue[*metrics](func() *metrics {
+	m := metrics{
+		reg:           prometheus.NewRegistry(),
+		serverMetrics: grpc_prometheus.NewServerMetrics(),
+	}
+	m.reg.MustRegister(m.serverMetrics)
+
+	f := promauto.With(m.reg)
+
+	m.newHashedRekordEntries = f.NewCounter(prometheus.CounterOpts{
+		Name: "rekor_new_hashedrekord_entries",
+		Help: "The total number of new dsse log entries",
+	})
+
+	m.newDsseEntries = f.NewCounter(prometheus.CounterOpts{
+		Name: "rekor_new_dsse_entries",
+		Help: "The total number of new dsse log entries",
+	})
+
+	m.httpLatency = f.NewHistogramVec(prometheus.HistogramOpts{
+		Name: "rekor_http_api_latency",
+		Help: "API Latency on HTTP calls",
+	}, []string{"code", "method"})
+
+	m.httpRequestsCount = f.NewCounterVec(prometheus.CounterOpts{
+		Name: "rekor_http_requests_total",
+		Help: "Count all HTTP requests",
+	}, []string{"code", "method"})
+
+	// TODO: add metrics from rekor v1 (anything but Counter appears to need to be a pointer)
```

Comment:
Do you want to do this in this PR or later?

Reply:
oh yeah... oops to be done later, I'll add an issue and include the issue number here